### PR TITLE
new primitive: selection

### PR DESCRIPTION
**Pull request by atk**
Selection inside textual inputs is simple; in contentEditable elements, it gets complicated. So simplification is a worthwile endeavor.

### CODE REVIEW

File: packages/selection/README.md [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+## Usage
+
+
+```ts
+import { createSelection } from "@solid-primitives/selection";
+
+const [selection, setSelection] = createSelection();
+
+selection(); // [HTMLDivElement, 1, 3]
+setSelection([document.querySelector('input'), 3, 3]);
```
Comment (thetarnav) [inline packages/selection/README.md:31]: I was confused by this initially. "Where did those values came from?"
```suggestion
selection(); // T: [HTMLElement, number, number]

setSelection([document.querySelector('input'), 1, 3]);
selection(); // => [<input/>, 1, 3]

// when no node is selected:
selection(); // => [null, NaN, NaN]
```

File: packages/selection/README.md [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+const [selection, setSelection] = createSelection();
+
+const ibanMask = anyMaskToFn('aa99999999999999999999');
+
+const inputMaskHandler = (ev) => {
+  const [node, start, end] = selection();
+  if (ev.currentTarget === node) {
+    const [value, selection] = ibanMask(node.innerHTML, [start, end]);
+    node.innerHTML = value;
+    setSelection(node, selection[0], selection[1]);
```
Comment (thetarnav) [inline packages/selection/README.md:49]: I believe you have to pass an array? *(I would prefer to have a ...rest option though)*
```suggestion
    setSelection([node, selection[0], selection[1]]);
```

File: packages/selection/src/index.ts [hunk 3]
```diff
@@ -0,0 +1,96 @@
+import { Accessor, createEffect, createSignal, onCleanup, Setter } from "solid-js";
+
+export type HTMLSelection = [node: HTMLElement | null, start: number, end: number];
+
+export const getTextNodes = (node: Node) => {
```
Comment (thetarnav) [inline packages/selection/src/index.ts:5]: If you are exporting this function, maybe would it be worth it to mention it in the readme, as an additional *vanilla* utility?

**Comment by thetarnav** [comment]
Is there an easy way to clear the selection?
The only way is to do `setSelection([null, NaN, NaN])` right?

File: packages/selection/README.md (hunk 2 above)
Comment (atk) [reply packages/selection/README.md:49]: Thanks, good catch.

File: packages/selection/src/index.ts (hunk 3 above)
Comment (atk) [reply packages/selection/src/index.ts:5]: I was exporting it mostly for testing purposes. If we're in need of such a function in other priitives, that's when I would move it to utils.

**Comment by atk** [comment]
> Is there an easy way to clear the selection? The only way is to do `setSelection([null, NaN, NaN])` right?

That's the way, yes. I should document that, too.

File: packages/selection/README.md (hunk 1 above)
Comment (atk) [reply packages/selection/README.md]: I added a few comments.

**Review by thetarnav — APPROVED** [review]
Looks great now :)

**Pull request merged**